slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -449,6 +459,16 @@ void SLAMServiceImpl::CreateMap() {
 
         map_builder.map_builder_->FinishTrajectory(trajectory_id);
         map_builder.map_builder_->pose_graph()->RunFinalOptimization();
+        auto local_poses = map_builder.GetLocalSlamResultPoses();
```

Comment:
initially pose was not being updated after an optimization occurs. Currently added this here for offline mode but not sure if there is a better place to include this

Reply:
I think this is fine!